sys/include/architecture.h

```diff
@@ -193,17 +197,20 @@ typedef uintptr_t   uinttxtptr_t;
 /**
  * @brief   Highest number an uword_t can hold
  */
-#define UWORD_MAX                   ((1ULL << ARCHITECTURE_WORD_BITS) - 1)
+#define UWORD_MAX                   ((1ULL << (ARCHITECTURE_WORD_BITS - 1)) - 1 \
+                                   + (1ULL << (ARCHITECTURE_WORD_BITS - 1)) )
```

Comment:
This I don't quite understand

Reply:
This is a workaround to avoid overflows.
Since `ULL` only guarantees to hold at least 64 bits, `1ULL << 64` can trigger
```
warning: left-shift-count >= width of type [-Wshift-count-overflow].
```

Reply:
I think `((unsigned long long)(-1LL))` should also work:

```
$ cat test.c                                           
#include <stdio.h>

int main(void)
{
    printf("%llu\n", (unsigned long long)(-1LL));
    return 0;
}
$ clang -Weverything -o test -Os test.c 
$ ./test                                                       
18446744073709551615
```

Reply:
(The original expression would exceed the range of `unsigned long long` on 64 bit architectures by shifting 64 bits to the left. That's why it needs to be adapted.)

Reply:
Yes, but `((unsigned long long)(-1LL))` would only work for 64 bit.

If you prefer something longer but more readable, we could just hardcode it like this
```
#if (ARCHITECTURE_WORD_BITS == 64)
#define UWORD_MAX (18446744073709551615ULL)
#else
#define UWORD_MAX ((1ULL << ARCHITECTURE_WORD_BITS) - 1)
#endif
```

Reply:
> Yes, but `((unsigned long long)(-1LL))` would only work for 64 bit.

Yes, `((uword_t)(-1LL))` would be the portable version of that.

Reply:
> > Yes, but `((unsigned long long)(-1LL))` would only work for 64 bit.
> 
> Yes, `((uword_t)(-1LL))` would be the portable version of that.

That is true, but we have the same problem with `SWORD_MIN` and `SWORD_MAX`.
Also, a subtle difference is that it changes the type.
(Which again could be migrated with `((unsigned long long)(uword_t)(-1LL))`.)

Reply:
Another option would be to move the defines up and hardcode all of them. Like this
```
#if DOXYGEN
/**
 * @brief   Highest number an uword_t can hold
 */
#define UWORD_MAX                  <NUM>
...
#elif (ARCHITECTURE_WORD_BITS == 64)
#define ARCHITECTURE_WORD_BYTES     (8U)
typedef uint64_t    uword_t;
typedef int64_t     sword_t;
#define UWORD_MAX   (UINT64_MAX)
#define SWORD_MIN   (INT64_MIN)
#define SWORD_MAX   (INT64_MAX)
#else
...
```

Reply:
but with the signed versions there should be an overflow with the shift, right?

Reply:
Both `1ULL << 64` and `1LL << 63` would overflow, that is why this code uses only `1ULL << 63` and `1LL << 62` and returns the correct result for 64 bit.

Reply:
I like the hardcoded 
but also 
`(0ULL - 1ULL)` s should just work

Reply:
I've added a fixup to hardcode all `{U,S}WORD_{MIN,MAX}` defines.
This also required adding the defines to the doxygen part.
For the doxygen values, I decided to add the corresponding equation in angle brackets, if you have other suggestions, let me know.
While I was at it, I also cleaned up the doxygen a bit. (There was a stray line in `PRIoSIZE` and the brief documentation `PRI{x,X}SIZE` was too long).